build-logic/src/main/kotlin/ktlint-publication.gradle.kts

```diff
@@ -7,13 +7,14 @@ plugins {
     signing
 }
 
-if (providers.gradleProperty("isKotlinDev").orNull.toBoolean()) {
-    val definedVersion = ext["VERSION_NAME"].toString().removeSuffix("-SNAPSHOT")
-    ext["VERSION_NAME"] = "$definedVersion-kotlin-dev-SNAPSHOT"
-}
+val definedVersion: String = providers.gradleProperty("VERSION_NAME").get()
 
-project.version = providers.gradleProperty("VERSION_NAME").orNull
-    ?: throw GradleException("Project version property is missing")
+project.version =
+    if (hasProperty("kotlinDev")) {
+        "${definedVersion.removeSuffix("-SNAPSHOT")}-kotlin-dev-SNAPSHOT"
+    } else {
+        definedVersion
+    }
```

Comment:
❗ 
Here's another thing I claim didn't work, but I might be wrong, I haven't found anything mentioning `isKotlinDev` property, so I assumed this should match `kotlinDev` property used elsewhere. 
Looking at the implementation, it concluded the intention was to publish artifacts using kotlinDev version to follow their own versioning, I updated the configuration and run `./gradlew publishToMavenLocal -PkotlinDev` to verify this will now follow previous naming convention:
<img width="758" alt="image" src="https://github.com/pinterest/ktlint/assets/36954793/6df73cec-81e7-4e9f-adfe-bfce2b9f97c4">

Please let me know if this is not intended, and I should rather remove the `isKotlinDev` if L10-13, or if it serves some other purpose

Reply:
I think that your reasoning is valid (see https://pinterest.github.io/ktlint/0.50.0/install/snapshot-build/#kotlin-development-version-snapshot). Maybe this got lost when rewriting the build from groovy to kotlin script. But, I don't  expect anyone ever has used the `kotlin-dev-SNAPSHOT`...